.github/workflows/build_test.yml

```diff
@@ -138,6 +161,7 @@ jobs:
   push-public:
     name: Build and Push Artifacts When Required
     runs-on: ubuntu-latest
+    if: ${{ needs.authorize.outputs.public_repo == 'true' }}
```

Comment:
I'm afraid this won't work. GitHub may not yet have the `needs.authorize.outputs.public_repo` available at the time it decides whether to run the job, even if it depends on the authorize. Thus, this job will never run. That's why it was decided to place this condition at the steps level.

Reply:
Reverted, thanks